### PR TITLE
Stream spec refactor

### PR DESCRIPTION
**Pull request by jedrzejkozal**
List of changes:
 - change iteration of StreamSpecification so instead of list of all classes return new and drifted classes separately (remove unused variable)
 - change class layout in the n_drifts case - it works better with longer task sequences
 - add max_classes_per_drift commandline argument
 - new unit tests

### CODE REVIEW

File: datasets/stream_spec.py [hunk 1]
```diff
@@ -30,57 +35,46 @@ def __init__(self, n_tasks: int, n_classes: int, random_seed: int = None,
...
-        if len(drift_indexes) < self.n_drifts:
-            drift_indexes.append(self.n_tasks-1)
+        drift_indexes = list(np.linspace(drift_duration, self.n_tasks, num=self.n_drifts, endpoint=False, dtype=int))
         print('Creating drifts at tasks: ', drift_indexes)
 
         classes_per_task = self.n_classes // self.n_tasks
-        last_class = 0
-        last_drifted_class = -1
-        task_classes = list()
+        drfit_begin_idx = 0
```
Comment (AlifAshrafee) [inline datasets/stream_spec.py]: Small typo here: `drfit_begin_idx` should be `drift_begin_idx `

**Review by AlifAshrafee — COMMENTED** [review]
Everything looks great to me. I have also run the test scripts to double-check the corner cases.
